File: springboot2.x-learning/springboot-data-jpa/src/main/java/com/rtxtitanv/model/Role.java
package com.rtxtitanv.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * @author rtxtitanv
 * @version 1.0.0
 * @name com.rtxtitanv.model.Role
 * @description 角色实体类
 * @date 2020/1/3 20:29
 */
@Setter
@Getter
@Entity
@Table(name = "role")
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "role_id")
    private Long roleId;
    @Column(name = "role_name")
    private String roleName;
    @Column(name = "role_type")
    private String roleType;
    @ManyToMany(mappedBy = "roles", cascade = CascadeType.PERSIST, fetch = FetchType.LAZY)
    private Set<Menu> menus = new HashSet<>();

    @Override
    public String toString() {
        return "Role{" +
                "roleId=" + roleId +
                ", roleName='" + roleName + '\'' +
                ", roleType='" + roleType + '\'' +
                '}';
    }
}

File: springboot2.x-learning/springboot-data-jpa-kotlin/src/main/kotlin/com/rtxtitanv/model/Role.kt
package com.rtxtitanv.model

import javax.persistence.*

/**
 * @name com.rtxtitanv.model.Role
 * @description 角色实体类
 * @author rtxtitanv
 * @date 2020/1/20 21:09
 * @version v1.0.0
 */
@Entity
@Table(name = "role")
class Role(@Id
           @GeneratedValue(strategy = GenerationType.IDENTITY)
           @Column(name = "role_id")
           var roleId: Long? = null,
           @Column(name = "role_name")
           var roleName: String? = null,
           @Column(name = "role_type")
           var roleType: String? = null,
           @ManyToMany(mappedBy = "roles", cascade = [CascadeType.PERSIST], fetch = FetchType.LAZY)
           var menus: MutableSet<Menu> = HashSet()) {

    override fun toString(): String {
        return "Role{" +
                "roleId=$roleId" +
                ", roleName='$roleName'" +
                ", roleType='$roleType'" +
                "}"
    }
}